src/lazy/text/buffer.rs

```diff
@@ -2183,6 +2248,26 @@ mod tests {
             );
         }
 
+        fn expect_match_location<'data, P, O>(&'data self, parser: P, expected_location: (usize, usize))
+        where
+            P: Parser<TextBuffer<'data>, O, IonParseError<'data>>,
+        {
+            let result = self.try_match(parser).unwrap_or_else(|e| {
+                panic!("Unexpected parse fail for input <{}>\n{e}", self.input)
+            });
+            let match_length = result.1;
+            // Inputs have a trailing newline and `0` that should _not_ be part of the match
```

Comment:
This was a hack from an old implementation of the text reader that couldn't properly detect incompleteness. It's no longer needed and can also be removed from where you copied it.